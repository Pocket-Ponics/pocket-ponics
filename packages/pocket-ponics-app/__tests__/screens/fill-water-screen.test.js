import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'

import FillWaterScreen from '../../screens/fill-water-screen'

describe('Fill Water Screen', () => {
	test('renders correctly', () => {
		const tree = shallow(
			<FillWaterScreen />
		).html()
		expect(tree).toMatchSnapshot()
	})

	test('continues registration', () => {
		const navigation = {
			navigate: jest.fn(),
			dispatch: jest.fn()
		}

		const tree = shallow(<FillWaterScreen navigation={navigation}/>)

		tree.find(TouchableOpacity).at(0).props().onPress()

		expect(navigation.dispatch).toHaveBeenCalled()
	})

	test('cancels registration', () => {
		const navigation = {
			navigate: jest.fn(),
			dispatch: jest.fn()
		}

		const tree = shallow(<FillWaterScreen navigation={navigation}/>)

		tree.find(TouchableOpacity).at(1).props().onPress()

		expect(navigation.navigate).toHaveBeenCalledWith('Greenhouse')
	})
})