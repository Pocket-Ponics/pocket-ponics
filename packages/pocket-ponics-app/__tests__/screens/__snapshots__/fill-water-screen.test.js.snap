// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fill Water Screen renders correctly 1`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Fill the water tank</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\">Unscrew the cap on the front of the water tank and pour in two gallons of water.  The water should be at the full tank mark.</Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Continue Setup</Text></View><View style=\\"align-items:center;opacity:1\\"><Text style=\\"color:#84AB71;font-size:22px\\">Cancel Setup</Text></View></View></View>"`;
